pkg/cmd/cli/cmd/process.go

```diff
@@ -111,122 +93,148 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 			}
 			return err
 		}
-
-		version, kind, err := mapper.VersionAndKindForResource("template")
-		if mapping, err = mapper.RESTMapping(kind, version); err != nil {
-			return err
-		}
+		templateObj.CreationTimestamp = util.Now()
+		infos = append(infos, &resource.Info{Object: templateObj})
 	} else {
-		obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		infos, err = resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 			NamespaceParam(namespace).RequireNamespace().
 			FilenameParam(filename).
 			Do().
-			Object()
-
+			Infos()
 		if err != nil {
 			return err
 		}
+	}
 
-		var ok bool
-		templateObj, ok = obj.(*api.Template)
+	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+
+	for i := range infos {
+		obj, ok := infos[i].Object.(*api.Template)
 		if !ok {
-			return fmt.Errorf("cannot convert input to Template: ", reflect.TypeOf(obj))
+			fmt.Fprintf(cmd.Out(), "unable to parse %q, not a valid Template but %s\n", obj.Name, reflect.TypeOf(obj))
+			continue
 		}
-		templateObj.CreationTimestamp = util.Now()
 
-		version, kind, err := kapi.Scheme.ObjectVersionAndKind(templateObj)
-		if err != nil {
-			return err
-		}
-		if mapping, err = mapper.RESTMapping(kind, version); err != nil {
-			return err
+		// If 'parameters' flag is set it does not do processing but only print
+		// the template parameters to console for inspection.
+		// If multiple templates are passed, this will print combined output for all
+		// templates.
+		if kcmdutil.GetFlagBool(cmd, "parameters") {
+			if len(infos) > 1 {
+				fmt.Fprintf(out, "\n%s:\n", obj.Name)
+			}
+			if err := describe.PrintTemplateParameters(obj.Parameters, out); err != nil {
+				fmt.Fprintf(cmd.Out(), "error printing parameters for %q: %v\n", obj.Name, err)
+			}
+			continue
 		}
-	}
 
-	if cmd.Flag("value").Changed {
-		injectUserVars(cmd, templateObj)
-	}
+		if label := kcmdutil.GetFlagString(cmd, "labels"); len(label) > 0 {
+			lbl, err := kubectl.ParseLabels(label)
+			if err != nil {
+				fmt.Fprintf(cmd.Out(), "error parsing labels: %v\n", err)
+				continue
+			}
+			if obj.ObjectLabels == nil {
+				obj.ObjectLabels = make(map[string]string)
+			}
+			for key, value := range lbl {
+				obj.ObjectLabels[key] = value
+			}
+		}
 
-	// If 'parameters' flag is set it does not do processing but only print
-	// the template parameters to console for inspection.
-	if kcmdutil.GetFlagBool(cmd, "parameters") == true {
-		err = describe.PrintTemplateParameters(templateObj.Parameters, out)
-		if err != nil {
-			return err
+		// Override the values for the current template parameters
+		// when user specify the --value
+		if cmd.Flag("value").Changed {
+			injectUserVars(cmd, obj)
 		}
-		return nil
-	}
 
-	label := kcmdutil.GetFlagString(cmd, "labels")
-	if len(label) != 0 {
-		lbl, err := kubectl.ParseLabels(label)
+		// TODO: use AsVersionedObjects to generate the runtime.Objects, because
```

Comment:
There should be no todo here - we don't transform templates.  If I asked you to add this comment I was wrong.